src/server/getters/get-profit-loss.ts

```diff
@@ -190,36 +232,85 @@ function bucketAtTimestamps<T extends Timestamped>(timestampeds: Dictionary<Arra
 
 function getProfitResultsForTimestamp(plsAtTimestamp: Array<ProfitLossTimeseries>, outcomeValuesAtTimestamp: Array<OutcomeValueTimeseries> | null): Array<ProfitLossResult> {
   const unsortedResults: Array<ProfitLossResult> = plsAtTimestamp.map((outcomePl) => {
-    const realized = outcomePl!.profit;
-    const outcome = outcomePl!.outcome;
-    const timestamp = outcomePl!.timestamp;
-    const marketId = outcomePl!.marketId;
-    let averagePrice = outcomePl!.price;
-    const minPrice = outcomePl!.minPrice;
-    const position = outcomePl!.position;
-
-    let unrealized = ZERO;
-    if (outcomeValuesAtTimestamp) {
-      const outcomeValue = outcomeValuesAtTimestamp[outcome].value.minus(minPrice);
-      const shareProfit = position.lt(ZERO) ? averagePrice.minus(outcomeValue) : outcomeValue.minus(averagePrice);
-      unrealized = shareProfit.multipliedBy(position.abs());
-    }
-
-    // Adjust display for scalars
-    averagePrice = averagePrice.plus(minPrice);
-
-    const total = realized.plus(unrealized);
+    const realizedCost = new Tokens(outcomePl.realizedCost);
+    const realizedProfit = new Tokens(outcomePl.profit);
+    const outcome = outcomePl.outcome;
+    const averagePricePerShareToOpenPosition = new Price(outcomePl.price);
+    const minPrice = new Price(outcomePl.minPrice);
+    const netPosition = new Shares(outcomePl.position);
+    const frozenFunds = new Tokens(outcomePl.frozenFunds);
+
+    // averagePrice is average price per share user paid to open this position;
+    // but, we want to display averagePrice; for scalar markets this means adding
+    // minPrice. Eg. if a user paid an average price of 4.5 for a scalar market, but
+    // the minPrice in that market is 3, we actually need to show 4.5+3=7.5 in the UI.
+    // TODO use new tradeDisplayPrice() from financialMath?
+    const averagePriceDisplay: Price = averagePricePerShareToOpenPosition.plus(minPrice);
+
+    const lastPrice: Price | undefined = outcomeValuesAtTimestamp ? new Price(outcomeValuesAtTimestamp[outcome].value).minus(minPrice) : undefined;
```

Comment:
Why undefined instead of null? Just curious.

Reply:
`undefined` seems more technically correct in my head. I pretty much never use null.